Acme.ProjectCompare/src/Acme.ProjectCompare.HttpApi/Samples/BookController.cs

```diff
@@ -6,65 +6,65 @@
-        {
-            var result = await _bookServices.GetBooks(pageSize,pageNumber,searchString);       
-            return Json(result);
-        }
-        [HttpGet("{id}")]
-        public async Task<IActionResult> GetBookById(Guid id)
-        {
-            var book = await _bookServices.GetBookById(id);
-            if(book == null)
+            private readonly IBookServices _bookServices;
+            public BookController(IBookServices bookServices)
             {
-                return null;
+                _bookServices = bookServices;
             }
-            return Ok(book);
-        }
-       
-        [HttpPost]
-        public async Task<IActionResult> CreateBook([FromBody] BookDto bookDto)
-        {
-            var result = await _bookServices.CreateBook(bookDto);
-            if(result == 0)
+            
+            [HttpGet]
+            public async Task<JsonResult> GetBooks([FromQuery] int pageSize,int pageNumber, string? searchString)
             {
-                return new BadRequestObjectResult(new { Message = "Update failed" });
-            }   
-            return Ok(new { Message = "Created Successfully" });
-        }
-        [HttpPut("{id}")]
-        public async Task<IActionResult> UpdateBook(Guid id, [FromBody] BookDto bookDto)
-        {
-            var result = await _bookServices.UpdateBook(id, bookDto);
-            if(result == 0)
+                var result = await _bookServices.GetBooks(pageSize,pageNumber,searchString);       
+                return Json(result);
+            }
+            [HttpGet("{id}")]
+            public async Task<IActionResult> GetBookById(Guid id)
             {
-                return new BadRequestObjectResult(new { Message = "Update failed" });
+                var book = await _bookServices.GetBookById(id);
+                if(book == null)
+                {
+                    return null;
+                }
+                return Ok(book);
             }
-            return Ok(new { Message = "Updated Successfully" });
-        }
-        [HttpDelete("{id}")]
-        public async Task<IActionResult> DeleteBook(Guid id)
-        {
-            var result = await _bookServices.DeleteBook(id);
-            if (result == 0)
+        
+            [HttpPost]
+            public async Task<IActionResult> CreateBook([FromBody] BookDto bookDto)
             {
-                return new BadRequestObjectResult(new { Message = "Delele Error" });
+                var result = await _bookServices.CreateBook(bookDto);
+                if(result == 0)
+                {
+                    return new BadRequestObjectResult(new { Message = "Update failed" });
+                }   
+                return Ok(new { Message = "Created Successfully" });
             }
-            return Ok(new { Message = "Deleted Successfully" });
+            [HttpPut("{id}")]
+            public async Task<IActionResult> UpdateBook(Guid id, [FromBody] BookDto bookDto)
```

Comment:
compare true or false when the result comes back from the service layer. (not 1 or 0)